lib/chatwithtools.py

```diff
@@ -113,8 +119,12 @@ def _process_single_input(self, input_data: dict[str, Any]) -> dict[str, Any]:
             message = json.loads(raw_message)
             if message['role'] == 'assistant':
                 if message['content'] == '':
-                    continue
-                messages.append(AIMessage(content=message['content']))
+                    if not message.get("tool_calls"):
+                        continue
+                    else:
+                        messages.append(AIMessage(content=generate_tool_call(message['tool_calls'][0])))
```

Comment:
What do you think of this refactor? It might be easier to read.
```py
if message['content'] == '' and message.get('tool_calls'):
    messages.append(AIMessage(content=generate_tool_call(message['tool_calls'][0])))
else:
    messages.append(AIMessage(content=message['content']))
```

Reply:
Good point, heh